components/concept/engine/src/main/java/mozilla/components/concept/engine/permission/PermissionRequest.kt

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.engine.permission
+
+/**
+ * Represents a permission request, used when engines need access to protected
+ * resources. Every request must be handled by either calling [grant] or [reject].
+ */
+interface PermissionRequest {
+    /**
+     * The origin URI which caused the permissions to be requested.
+     */
+    val uri: String
```

Comment:
nit: This may not be 100% correct for an App Permission. They are not for a specific origin and they may not be caused by the "current URL". I think it's okay to have this for the sake of having a shared permission request interface - however I wonder whether we should make this nullable and have this set to null for App Permissions.

Reply:
I will take a look!